File: design/design_dialogs/src/main/java/ru/tensor/sbis/design_dialogs/dialogs/content/ContentCreator.java
package ru.tensor.sbis.design_dialogs.dialogs.content;

import java.io.Serializable;

/**
 * Наследник {@link BaseContentCreator} с объявлением наследования от {@link Serializable}
 * Класс, реализующий данный интерфейс, должен быть неанонимным для того,
 * чтобы поддержать функционал {@link Serializable}.
 *
 * @author sa.nikitin
 */
public interface ContentCreator extends BaseContentCreator, Serializable {
}

File: design/container/src/main/java/ru/tensor/sbis/design/container/ContentCreator.kt
package ru.tensor.sbis.design.container

/**
 * Фабрика для создания контента, если контент реализует Parcelable контейнер будет восстанавливаться при повороте
 * @author ma.kolpakov
 */
interface ContentCreator<out CONTENT : Content> {
    fun createContent(): CONTENT
}